test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/lang/MockCharSequence.scala

```diff
@@ -0,0 +1,30 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js Test Suite        **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013-2017, LAMP/EPFL   **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+package org.scalajs.testsuite.javalib.lang
+
+object MockCharSequence {
```

Comment:
Call this a Fake... or a Dummy... or even WrappedStringCharSequence. A mock is something that allows to check if methods have been called.
See: https://stackoverflow.com/questions/346372/whats-the-difference-between-faking-mocking-and-stubbing